core/src/main/scala/com/avast/clients/rabbitmq/converters.scala

```diff
@@ -21,6 +21,14 @@ trait CheckedDeliveryConverter[A] extends DeliveryConverter[A] {
 
 object DeliveryConverter {
   implicit val identity: DeliveryConverter[Bytes] = (b: Bytes) => Right(b)
+  implicit val bytesArray: DeliveryConverter[Array[Byte]] = (b: Bytes) => Right(b.toByteArray)
+  implicit val utf8String: CheckedDeliveryConverter[String] = new CheckedDeliveryConverter[String] {
+    override def canConvert(d: Delivery[Bytes]): Boolean = d.properties.contentType match {
+      case Some(contentType) => contentType.toLowerCase.contains("charset=utf-8") || contentType.toLowerCase.startsWith("text")
```

Comment:
I still don't like this because IMHO it matches many crappy things that could come. I propose to use regex, sth like `text/[a-zA-Z]; ?charset=utf\-8` (depends on what RFC says).

Reply:
The point of this current code is to accept all `text` MIME types, and also all other `Content-Type`s with `charset` set to `utf-8`. And this seems to be correct to me.

Do you have any common counterexample, please?

Or, we could rework the `utf8String` to be just plain `DeliveryConverter` if we are not able to agree on it. Because then we can suppose it would be unclear for users as well.

Reply:
Please change it to support `charset="utf-8"` and it will be good enough I guess.

I can imagine even extracting the charset from the property value and I'm also willing to help you but don't have time to solve it right now.  
My current working version is:
```scala
import java.nio.charset.Charset
import scala.util.Try

object Regexes extends App {
  import java.util.regex.Pattern

  private val CONTENT_TYPE_REGEX = "([ |\t]*content-type[ |\t]*:)(.*)"
  private val CONTENT_TYPE_PATTERN = Pattern.compile(CONTENT_TYPE_REGEX, Pattern.CASE_INSENSITIVE)
  private val CHARSET_REGEX = "['|\"]?([^\"^'^;^,]*)['|\"]?"
  private val CHARSET_PATTERN = Pattern.compile(CHARSET_REGEX, Pattern.CASE_INSENSITIVE)

  val matcher = CONTENT_TYPE_PATTERN.matcher("Content-Type: text/plain; charset=utf-8")

  if (matcher.matches()) {
    val value = matcher.group(2)
    val attrs: Map[String, String] = value
      .split(";")
      .tail
      .map(_.trim)
      .flatMap(_.split("=") match {
        case Array(k, v) => Some(k -> v)
        case _ => None
      })
      .toMap

    val charset: Option[Charset] = attrs
      .get("charset")
      .flatMap { v =>
        val matcher = CHARSET_PATTERN.matcher(v)

        if (matcher.matches()) {
          Option(matcher.group())
        } else None
      }
      .flatMap(cn => Try(Charset.forName(cn)).toOption)

    attrs
  }

}
```
It's able to  parse the header/property and extract it's parameters (see https://www.w3.org/Protocols/rfc1341/4_Content-Type.html) so it works and it's just necessary to somehow wrap it and make it usable.  
Feel free to use the code, merge this after the small change I require above or wait for me to finish this improvement :wink:

Reply:
Are you sure "starts with `text`" **OR** "contains `charset=utf-8` is correct? :thinking: Sorry, didn't notice the operator since now.

Reply:
I've just made the change you wanted.

Btw. it is no problem to extract the charset from `Content-Type`. The show-stopper is that `canConvert` method accepts `Delivery[Bytes]` BUT the `convert` method accepts just plain `Bytes`. So we can extract the current `Charset` in `canConvert` method but we cannot access this `Charset` inside the `convert` method.

Reply:
Yes, the `OR` is intentional because the `charset` is not required.

Reply:
Sorry. I've forgotten this small sad fact again. OK, it has to be fixed for UTF-8 then.

```scala
contentType.toLowerCase.startsWith("text") && (
  contentType.toLowerCase.contains("charset=utf-8") ||
  contentType.toLowerCase.contains("charset=\"utf-8\"")
)

```
?

Reply:
I've just pushed slightly different code. My intention is to be able to accept also `Content-Type` without charset specified (because it's not obligatory to send it).

Reply:
I've also renamed `utf8String` to `utf8Text` so it's more clear that `text` `Content-Type` is required.